diff --git a/clientside_validation_webform/clientside_validation_webform.module b/clientside_validation_webform/clientside_validation_webform.module
index 58893c1..ae6c8b9 100755
--- a/clientside_validation_webform/clientside_validation_webform.module
+++ b/clientside_validation_webform/clientside_validation_webform.module
@@ -118,7 +118,7 @@ function clientside_validation_webform_after_build_recurse($form_id, &$form, &$f
           }
         }
       }
-      if (isset($element['#type']) && $element['#type'] == 'captcha') {
+      if (isset($element['#type']) && $element['#type'] == 'captcha' && empty($element['captcha_widgets']['captcha_response']['#exclude_cv'])) {
         $settings = clientside_validation_settings_current_form();
         if ($settings['validate_options']['captcha']) {
           $title = _clientside_validation_set_title(isset($element['captcha_widgets']['captcha_response']['#title']) ? $element['captcha_widgets']['captcha_response']['#title'] : $element['#name']);
