diff --git a/docroot/modules/file/file.module b/docroot/modules/file/file.module
index e7ff997..39f94c4 100644
--- a/docroot/modules/file/file.module
+++ b/docroot/modules/file/file.module
@@ -239,11 +239,10 @@ function file_ajax_upload() {
   $form_build_id = (string) array_pop($form_parents);
 
   if (empty($_POST['form_build_id']) || $form_build_id != $_POST['form_build_id']) {
-    // Invalid request.
-    drupal_set_message(t('An unrecoverable error occurred. The uploaded file likely exceeded the maximum file size (@size) that this server supports.', array('@size' => format_size(file_upload_max_size()))), 'error');
-    $commands = array();
-    $commands[] = ajax_command_prepend(NULL, theme('status_messages'));
-    return array('#type' => 'ajax', '#commands' => $commands);
+    echo '<script type="text/javascript">alert("' . t('The uploaded file likely exceeded the maximum file size (@size) that this server supports.\r\nPlease upload a new file that corresponds to maximum allowed size.', array('@size' => format_size(file_upload_max_size()))) . '");</script>';
+    list($form, $form_state, $form_id, $form_build_id, $commands) = ajax_get_form();
+    drupal_rebuild_form($form_id, $form_state);
+    return;
   }
 
   list($form, $form_state, $form_id, $form_build_id, $commands) = ajax_get_form();
